src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ReferralRepositoryTest.kt

```diff
@@ -1,59 +1,312 @@
+      assertThat(summaries.size).isEqualTo(2)
+      assertThat(contains(summaries, expectedSummary(referralWithChangedAssignee)))
+      assertThat(contains(summaries, expectedSummary(referralWithMultipleChangedAssignee)))
+      summaries.contains(expectedSummary(referralWithChangedAssignee))
+    }
+
+    @Test
+    fun `will not exclude referral when not assigned`() {
+      val referralWithNoAssignee = createReferral(true, 0)
+
+      val summaries = referralRepository.referralSummaryForServiceProviders(listOf(referralWithNoAssignee.serviceProvider.id))
 
       assertThat(summaries.size).isEqualTo(1)
-      assertThat(summaries.get(0).referenceNumber).isEqualTo("JS18726AC")
-      assertThat(summaries.get(0).interventionTitle).isEqualTo("Sheffield Housing Services")
-      assertThat(summaries.get(0).assignedToUserName).isEqualTo("bernard.beaks")
-      assertThat(summaries.get(0).serviceUserFirstName).isEqualTo("firstName")
-      assertThat(summaries.get(0).serviceUserLastName).isEqualTo("lastName")
+      assertThat(summaries[0].assignedToUserName).isNull()
+      assertThat(contains(summaries, expectedSummary(referralWithNoAssignee)))
     }
   }
+
+  @Nested
+  inner class StatusExclusion {
+
+    @Test
+    fun `will exclude cancelled referrals`() {
+      val referral = createEndedReferral(
+        true,
+        endRequestedAt = OffsetDateTime.now(),
+        concludedAt = OffsetDateTime.now(),
+        eosR = null,
+      )
+
+      val summaries = referralRepository.referralSummaryForServiceProviders(listOf(referral.serviceProvider.id))
+
+      assertThat(summaries.size).isEqualTo(0)
+    }
+
+    @Test
+    fun `can obtain those requested to be ended and not concluded`() {
+      val referral = createEndedReferral(
+        true,
+        endRequestedAt = OffsetDateTime.now(),
+        concludedAt = null,
+        eosR = null,
+      )
+
+      val summaries = referralRepository.referralSummaryForServiceProviders(listOf(referral.serviceProvider.id))
+
+      assertThat(summaries.size).isEqualTo(1)
+      assertThat(contains(summaries, expectedSummary(referral)))
+    }
+
+    @Test
+    fun `can obtain prematurely or completely ended referral`() {
+      val referral = createEndedReferral(
+        true,
+        endRequestedAt = null,
+        concludedAt = OffsetDateTime.now(),
+        eosR = endOfServiceReport.create(),
+      )
+
+      val summaries = referralRepository.referralSummaryForServiceProviders(listOf(referral.serviceProvider.id))
+
+      assertThat(summaries.size).isEqualTo(1)
+      assertThat(contains(summaries, expectedSummary(referral)))
+    }
+
+    @Test
+    fun `will exclude referral if draft`() {
+      val referralWithNoAssignee = createDraftReferral(true)
+
+      val summaries = referralRepository.referralSummaryForServiceProviders(listOf(referralWithNoAssignee.serviceProvider.id))
+
+      assertThat(summaries.size).isEqualTo(0)
+    }
+  }
+
+  private fun createReferral(
```

Comment:
why don't we have this in the factory classes, as it's what they're there for. Could be useful elsewhere.

Reply:
Where should these go?